Proposal for making a SPID compliant set of templates

brunato opened the pull request:
This proposal try to get a uniform template structure for app, using a base template (`spid_base.html`) and a SPID compliant template for errors.

Almost all templates have to be modified in a real project, but still having a minimal common structure reduces the work needed for adaptation. Inner templates (currently only `wayf.html` and `spid_login_error.html`) probably will need only slight modifications, and still have to use part of _bootstrap-italia_ CSS rules.

The base template usually will requires more adaptation using the styles of the hosting site (navigation bar, colors, etc.).

This is only an initial step for making a final template set for the app and requires other decisions (eg. do i use the base template also for attribute echo?) and modifications (eg. `spid_login_error.html` needs a better layout).

The `wayf.html` template and the _templatetags_ have been moved to the reusable app, because it's the simplest way of making them available in a real installation. Otherwise they can be moved again (almost the base template and all the ones that use it) to the demo part, if preferred. 

Note: `wayf.html` of the demo app has been temporarily renamed to `wayf_.html` to avoid the overriding of the new version.

Thanks

brunato committed:
Remove Django 3.0 tests in tox.ini

brunato committed:
Clean saml2 subpackage imports in src/djangosaml2_spid/views.py

brunato committed:
Move demo project's templatetags to app

  - These templatetags are needed for validate an installation

brunato committed:
Add SPID base template and wayf.html for reusable app

peppelinux commented on src/djangosaml2_spid/templates/spid_base.html:
```diff
@@ -0,0 +1,364 @@
+{% load static %}
+
+<!DOCTYPE html>
+<html lang="it">
```
https://github.com/italia/design-django-theme

we could get it as dependency and have a spid_base that only inherit the desidered blocks.
TODO now or in a future release. If the second we should open an issue related to a milestone

brunato replied:
I don't think that adding this as a dependency is a good idea. For me in Django this is a choice that can be better accomplished individually at project level (eventually we can add an advice and the link for encouraging this ...).

At app level i think that is important to provide the leaf templates (wayf.html, spid_error.html and so on) as final and compliant as possible for SPID. The spid_base.html will be in any case overridden in real installations, also with the option of building one that uses _design-django-theme_'s base template.

To simplify this operation we can change the block names currently defined in _spid_base.html_ as:

- `head-extra-scripts` -> `extra_head`
- `container` (does not change ...)
- `body-extra-scripts` -> `extra_scripts`

peppelinux replied:
LGTM

peppelinux commented on example/spid_config/metadata/spid-sp-test.xml:
```diff
@@ -0,0 +1,23 @@
+<ns0:EntityDescriptor xmlns:ns0="urn:oasis:names:tc:SAML:2.0:metadata" xmlns:ns1="http://www.w3.org/2000/09/xmldsig#" entityID="http://localhost:8080"><ns0:IDPSSODescriptor protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol" WantAuthnRequestsSigned="false"><ns0:KeyDescriptor use="signing"><ns1:KeyInfo><ns1:X509Data><ns1:X509Certificate>MIIEGDCCAwCgAwIBAgIJAOrYj9oLEJCwMA0GCSqGSIb3DQE ...
```
is this required?

brunato replied:
Not really, maybe it's a typo. I'm going to remove it with another commit that also changes the block names.

brunato committed:
Remove unecessary metadata files from demo and tests

brunato committed:
Rename SPID base template blocks

  - Use italia/design-django-theme compatible names:

    head-extra-scripts -> extra_head
    body-extra-scripts -> extra_scripts

peppelinux commented:
A test failed..  .?

brunato commented:
I removed also `tests/metadata/spid-sp-test.xml`, that was checked by a test (test_default_spid_saml_config), so i modified this test to avoid failures ... (the CI test 3.9+master fails for other internal failure IMO)
